web/src/actions/apd.js

```diff
@@ -221,10 +222,75 @@ export const submitAPD = (save = saveApd) => (dispatch, getState) =>
     .then(() => {
       dispatch({ type: SUBMIT_APD_REQUEST });
 
-      const { apd: { data: apd } } = getState();
+      const { apd: { data: { id: apdID } }, budget } = getState();
+      const getFederalShareByFFYQuarter = fundingSource =>
+        Object.entries(budget.quarterly[fundingSource]).reduce(
+          (accum, [ffy, quarters]) => ({
+            ...accum,
+            [ffy]: Object.entries(quarters).reduce(
+              (qAccum, [quarter, percent]) => ({
+                ...qAccum,
+                [quarter]: {
+                  percent,
+                  contractors:
+                    budget[fundingSource].contractors[ffy].federal *
+                    percent /
+                    100,
+                  expenses:
+                    budget[fundingSource].expenses[ffy].federal * percent / 100,
+                  statePersonnel:
+                    budget[fundingSource].statePersonnel[ffy].federal *
+                    percent /
+                    100,
+                  total:
+                    budget[fundingSource].combined[ffy].federal * percent / 100
+                }
+              }),
+              {
+                subtotal: {
+                  contractors: budget[fundingSource].contractors[ffy].federal,
+                  expenses: budget[fundingSource].expenses[ffy].federal,
+                  statePersonnel:
+                    budget[fundingSource].statePersonnel[ffy].federal,
+                  total: budget[fundingSource].combined[ffy].federal
+                }
+              }
+            )
+          }),
+          {
+            total: {
+              contractors: budget[fundingSource].contractors.total.federal,
+              expenses: budget[fundingSource].expenses.total.federal,
+              statePersonnel:
+                budget[fundingSource].statePersonnel.total.federal,
+              total: budget[fundingSource].combined.total.federal
+            }
+          }
+        );
+
+      const tables = {
+        summaryBudgetTable: {
+          hie: budget.hie,
+          hit: budget.hit,
+          mmis: budget.mmis,
+          total: budget.combined
+        },
+        federalShareByFFYQuarter: {
+          hitAndHie: getFederalShareByFFYQuarter('hitAndHie'),
```

Comment:
Good point.  We should move this to the reduce and update this action and the QuarterlyBudgetSummary component to just use the state data rather than do this math itself.  Created #762 for this.